src/routers/xcm/Router.sol

```diff
@@ -25,13 +26,63 @@ contract ConnectorXCMRouter {
     using TypedMemView for bytes29;
     using ConnectorMessages for bytes29;
 
+    /// --- Properties ---
     ConnectorLike public immutable connector;
+    address immutable centrifugeChainOrigin;
+    bytes centrifugeChainHandleCallIndex;
+    XcmWeightInfo xcmWeightInfo;
 
-    address centrifugeChainOrigin;
+    /// --- Storage ---
+    /// Auth storage
+    mapping(address => uint256) public wards;
 
-    constructor(address connector_, address centrifugeChainOrigin_) {
+    /// --- Events ---
+    event Rely(address indexed user);
+    event Deny(address indexed user);
+    event File(bytes32 indexed what, XcmWeightInfo xcmWeightInfo);
+
+    // Types
+    struct XcmWeightInfo {
+        // The weight limit in Weight units we accept amount to pay for the
+        // execution of the whole XCM on the Centrifuge chain. This should be
+        // transactWeightAtMost + an extra amount to cover for the other
+        // instructions in the XCM message.
+        uint64 buyExecutionWeightLimit;
+        // The weight limit in Weight units we accept paying for having the Transact
+        // call be executed. This is the cost associated with executing the handle call
+        // in the Centrifuge.
+        uint64 transactWeightAtMost;
+        // The amount to cover for the fees. It will be used in XCM to buy
+        // execution and thus have credit for pay those fees.
+        uint256 feeAmount;
+    }
+
+    constructor(address connector_, address centrifugeChainOrigin_, bytes memory centrifugeChainHandleCallIndex_) {
         connector = ConnectorLike(connector_);
         centrifugeChainOrigin = centrifugeChainOrigin_;
+        centrifugeChainHandleCallIndex = centrifugeChainHandleCallIndex_;
+        xcmWeightInfo = XcmWeightInfo({
+            buyExecutionWeightLimit: 19000000000,
+            transactWeightAtMost: 8000000000,
+            feeAmount: 1000000000000000000
+        });
+        wards[msg.sender] = 1;
+    }
+
+    /// -- Auth ---
+    modifier auth() {
+        require(wards[msg.sender] == 1, "ConnectorXCMRouter/not-authorized");
+        _;
+    }
+
+    function rely(address usr) external auth {
```

Comment:
is `auth` the retun valiue here?

Reply:
nop! `auth` is a modifier (in Solidity language), which really means it is a guard; so `rely ` is only executed if `auth` doesn't fail. Like a higher-level function basically;